### PR TITLE
created file for abstract metric class

### PR DESCRIPTION
**Pull request by alin5201314**
added get metric method and get weighted sum method

### CODE REVIEW

File: metric_superclass [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//Metric superclass with get_weighted_sum and get _metric metric methods
+
+abstract class metric{
+    private repo: Repository;
+
+    abstract get_metric(repo: Repository);
+
+    get_weigted_sum(license:number,BusFactor:number,Correctness:number,RampUp :number,ResponsiveMaintainer:number): number{
```
Comment (priyankapereira1) [inline metric_superclass:8]: typo - weighted sum

**Comment by priyankapereira1** [comment]
Okay - Robert already made changes in the metric.ts file with the abstract Metric class.
The new function get_weighted_sum is important but I don't think the flow of our class structure is correct - maybe we need to discuss? I can get back to you.

I think that function you made get_weighted_sum can just be pasted into the metric.ts file and this file can be deleted.

**Comment by rsego-hub** [comment]
Is it necessary to move the whole metric superclass to a different file? Are we doing it for managing variable access? I was under the impression we wanted to have the superclass and subclass in one file so that get_weighted_sum can access the subclasses easily and so that the subclasses can extend the superclass without using an import. Yeah we should probs discuss.

**Comment by priyankapereira1** [comment]
Closing due to design changes by team members - this was not the original plan anyway.

**Pull request closed without merging**